pandas/core/arrays/floating.py

```diff
@@ -0,0 +1,636 @@
+import numbers
+from typing import TYPE_CHECKING, List, Optional, Tuple, Type, Union
+import warnings
+
+import numpy as np
+
+from pandas._libs import lib, missing as libmissing
+from pandas._typing import ArrayLike, DtypeObj
+from pandas.compat import set_function_name
+from pandas.compat.numpy import function as nv
+from pandas.util._decorators import cache_readonly
+
+from pandas.core.dtypes.cast import astype_nansafe
+from pandas.core.dtypes.common import (
+    is_bool_dtype,
+    is_datetime64_dtype,
+    is_float,
+    is_float_dtype,
+    is_integer,
+    is_integer_dtype,
+    is_list_like,
+    is_object_dtype,
+    pandas_dtype,
+)
+from pandas.core.dtypes.dtypes import register_extension_dtype
+from pandas.core.dtypes.missing import isna
+
+from pandas.core import nanops, ops
+from pandas.core.array_algos import masked_reductions
+from pandas.core.ops import invalid_comparison
+from pandas.core.ops.common import unpack_zerodim_and_defer
+from pandas.core.tools.numeric import to_numeric
+
+from .masked import BaseMaskedArray, BaseMaskedDtype
+
+if TYPE_CHECKING:
+    import pyarrow  # noqa: F401
+
+
+class FloatingDtype(BaseMaskedDtype):
+    """
+    An ExtensionDtype to hold a single size of floating dtype.
+
+    These specific implementations are subclasses of the non-public
+    FloatingDtype. For example we have Float32Dtype to represent float32.
+
+    The attributes name & type are set when these subclasses are created.
+    """
+
+    name: str
+    base = None
+    type: Type
+
+    def __repr__(self) -> str:
```

Comment:
should e in the BaseMaskedArray class

Reply:
> should be in the BaseMaskedArray class

There would be nothing else using it, as the Integer/Boolean also have custom implementations. It needs some refactor to be able to share.